### PR TITLE
add non blocking case rate limits

### PR DESCRIPTION
**Pull request by calellowitz**
## Technical Summary

This adds basic rate limiting for case transactions (as triggered by forms). I set the limits to 10x the form ones, which I think are probably way too high (if form limits aren't being hit, it would be shocking if these case limits are), but I wanted to open this now for review on the approach itself. It is heavily coupled to the existing form limits, being part of the same check. It functions similarly to the limits in place for system forms and exempted domains, in that it will not block submission, just potentially delay it, and most importantly send info to datadog when that occurs.

I will add tests before we merge, but didn't want review to have to wait on them.

### Safety story

This has no user facing effects (just sends a metric when the limit is hit), with a one line modification to form processing code, which itself is heavily tested.

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/receiverwrapper/rate_limiter.py [hunk 1]
```diff
@@ -67,6 +67,45 @@ def _get_per_user_submission_rate_definition(domain):
 )
 
 
+global_case_rate_limiter = RateLimiter(
+    feature_key='global_case_updates',
+    get_rate_limits=lambda: get_dynamic_rate_definition(
+        'global_case_updates',
+        default=RateDefinition(
+            per_hour=170000,
```
Comment (snopoke) [inline corehq/apps/receiverwrapper/rate_limiter.py:75]: if these are meant to be relative to the form limits could we make those a constant: 

`FORMS_PER_HOUR_GLOBAL * 10`

**Review by snopoke — COMMENTED** [review]
Some questions:

* Can you articulate what the goal for this is? Are we wanting to protect against external forms or internal? Just checking because internal forms don't get rate limited nearly as much due to the defaults in `submit_form_locally`.
* 10 x forms seems fine for normal submissions but our bulk submissions from imports & case rules etc are 100 cases per form. How do you see this impacting those forms?
*

**Review by snopoke — APPROVED** [review]
Thanks for the feedback.

**Review by dannyroberts — APPROVED** [review]
Looks good to me. Something I often do for metrics-only changes like this is test it on staging, and just make sure the metric works/looks the way you think it should (while also double-checking that the change doesn't cause an error).

**Comment by snopoke** [comment]
> Looks good to me. Something I often do for metrics-only changes like this is test it on staging, and just make sure the metric works/looks the way you think it should (while also double-checking that the change doesn't cause an error).

Just to be clear, this is not a metrics only change correct? This will indeed delay form submissions if the limits are hit (e.g. if a case update rule matches 3M cases for example).

**Comment by dannyroberts** [comment]
@snopoke you're right, thanks for pointing that out. I'd gotten it mixed up from a conversation I'd had with Cal earlier. It would be nice to be able to get the metrics before having it take any action whatsoever, which I think you can do by changing the added call to use something like `max_wait=0, delay_rather_than_reject=False` (it will quick "reject", and then ignore the rejection). Alternatively, you could make the global or domain-level limit so high that it can't possibly be reached, and then lower it as you have data about what's typical usage. That said, I think the risk as is is fairly low as well, and these thresholds can be quickly changed.

**Comment by snopoke** [comment]
> @snopoke you're right, thanks for pointing that out. I'd gotten it mixed up from a conversation I'd had with Cal earlier. It would be nice to be able to get the metrics before having it take any action whatsoever, which I think you can do by changing the added call to use something like `max_wait=0, delay_rather_than_reject=False` (it will quick "reject", and then ignore the rejection). Alternatively, you could make the global or domain-level limit so high that it can't possibly be reached, and then lower it as you have data about what's typical usage. That said, I think the risk as is is fairly low as well, and these thresholds can be quickly changed.

I wasn't suggesting we change it, just confirming my understanding. I think the limits are high enough that they won't impact normal operations.

**Comment by calellowitz** [comment]
@dannyroberts I would like to keep them the way they are (high but able to delay if we really do hit them), unless you have strong objections, in which case I can make them truly metrics only.

**Comment by dannyroberts** [comment]
Hey @calellowitz, no objections. I was just thinking aloud. I agree the limits are high enough, so I'm happy with this as-is

**Pull request merged**